components/push/subscriber/src/lib.rs

```diff
@@ -66,7 +68,15 @@ impl PushManager {
         record.app_server_key = self.config.vapid_key.clone();
         record.native_id = Some(reg_token);
         self.store.put_record(&record)?;
-        // TODO: just return Record
+        // store the meta information if we've not yet done that.
+        if self.store.get_meta("uaid")?.is_none() {
```

Comment:
Should we (do we?) have a transaction open here? I guess we only have a single connection, and so it shouldn' tmatter.

Reply:
ideally, there would only be one "uaid". There's been some discussion about how this would be handled by containers (if/when they're brought over to mobile), but even then, having a single UAID may be the answer. UAID is more about delivery to a given device, channel about segregating to recipient processes. Same is true with the registration ID, to an extent, but that's generally initialized at app start and handed to us. The meta data is more a write once, read many store.